### PR TITLE
fix: forward context in UserSignupCleanup controller

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: controllers/usersignupcleanup/usersignup_cleanup_controller.go [hunk 1]
```diff
@@ -82,7 +81,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 			if createdTime.Time.Before(unverifiedThreshold) {
 				reqLogger.Info("Deleting UserSignup due to exceeding unverified retention period")
-				return reconcile.Result{}, r.DeleteUserSignup(instance, reqLogger)
+				return reconcile.Result{}, r.DeleteUserSignup(ctx, instance)
```
[the remark this answers is not in the document]
Comment (filariow) [reply controllers/usersignupcleanup/usersignup_cleanup_controller.go:84]: As we are passing the context and not performing any customization on the reqLogger, I thought it would be good in the `DeleteUserSignup` function to retrieve the logger from context (cf. https://github.com/codeready-toolchain/host-operator/blob/a93ab2f2fb363e6783b6f73946145ca9ad7b70e6/controllers/usersignupcleanup/usersignup_cleanup_controller.go#L165). As a result the username will still be logged.

[the remark this answers is not in the document]
Comment (alexeykazakov) [reply controllers/usersignupcleanup/usersignup_cleanup_controller.go:84]: But the logger in the context is a different instance. It's not the one we create at `reqLogger = reqLogger.WithValues("username", instance.Spec.Username)`, right?

[the remark this answers is not in the document]
Comment (filariow) [reply controllers/usersignupcleanup/usersignup_cleanup_controller.go:84]: Oh, I missed that. Thanks for pointing it out. Let me fix that.

[the remark this answers is not in the document]
Comment (filariow) [reply controllers/usersignupcleanup/usersignup_cleanup_controller.go:85]: I added a line to update the logger in the context. Next invocations of `log.FromContext(ctx)` will get the logger with values set. WDYT?

https://github.com/codeready-toolchain/host-operator/blob/dd35d0bf24133996c269d84395b86fb08e1dd973/controllers/usersignupcleanup/usersignup_cleanup_controller.go#L66

**Review by MatousJobanek — APPROVED** [review]
@filariow Thanks a lot for this fix. 
But could you please stop using force-push on PRs? It's very hard to follow the changes you made. Just keep pushing new commits to the PRs, the commits will be then squashed when the PR is being merged.

**Review by mfrancisc — APPROVED** [review]
Nice job 👍

**Review by alexeykazakov — APPROVED** [review]
